File: updated/lagomvinkladeharder/submissions/time_limit_exceeded/sol2.cpp
#include <cstdio>
#include <cmath>
#include <set>
#include <algorithm>
#include <tuple>
#include <vector>
using namespace std;

int main() {
	int n;
	while (scanf("%d", &n)) {
		int res = 0;
		for(int i=1; i<=n; i++) {
			for(int j=i; j<=n; j++) {
				int k = (int)round(sqrt(i*i + j*j - i*j));
				if (k*k == i*i + j*j - i*j && k <= n && k > 0) {
					res++;
				}
			}
		}

		printf("%d\n", res);
	}
	return 0;
}

File: do_not_install/lagomvinkladehard/submissions/accepted/sol2.py
import math
n = int(input())
res = 0
for i in range(1, n+1):
    for j in range(i, n+1):
        k = round(math.sqrt(i*i + j*j - i*j))
        if k*k == i*i + j*j - i*j and k <= n:
            res += 1

print(res)
